From 9a33b3e37735dfa34b01b1a82a017c891f43ebb1 Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Wed, 11 Jul 2012 21:08:27 +0000
Subject: [PATCH] Test getContent() and getContent().getContentInfo() in
 isolation for VFS-427: NPE on HttpFileObject.getContent().getContentInfo().
 No issue yet.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/vfs/trunk@1360399 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/vfs2/test/ProviderReadTests.java  | 29 +++++++++++++++++--
 1 file changed, 27 insertions(+), 2 deletions(-)

diff --git a/core/src/test/java/org/apache/commons/vfs2/test/ProviderReadTests.java b/core/src/test/java/org/apache/commons/vfs2/test/ProviderReadTests.java
index 76c5f2b8c7..12676f7051 100644
--- a/core/src/test/java/org/apache/commons/vfs2/test/ProviderReadTests.java
+++ b/core/src/test/java/org/apache/commons/vfs2/test/ProviderReadTests.java
@@ -25,6 +25,8 @@
 import junit.framework.Assert;
 
 import org.apache.commons.vfs2.Capability;
+import org.apache.commons.vfs2.FileContent;
+import org.apache.commons.vfs2.FileContentInfo;
 import org.apache.commons.vfs2.FileObject;
 import org.apache.commons.vfs2.FileSystem;
 import org.apache.commons.vfs2.FileSystemException;
@@ -140,7 +142,7 @@ protected void assertSameStructure(final FileObject folder, final FileInfo expec
     public void testType() throws Exception
     {
         // Test a file
-        FileObject file = getReadFolder().resolveFile("file1.txt");
+        FileObject file = resolveFile1Txt();
         assertSame(FileType.FILE, file.getType());
         assertTrue(file.isFile());
 
@@ -241,12 +243,17 @@ public void testFolderIsReadable() throws Exception
         Assert.assertTrue(folder.isReadable());
     }
     
+    private FileObject resolveFile1Txt() throws FileSystemException
+    {
+        return getReadFolder().resolveFile("file1.txt");
+    }
+    
     /**
      * Tests can perform operations on a folder while reading from a different files.
      */
     public void testConcurrentReadFolder() throws Exception
     {
-        final FileObject file = getReadFolder().resolveFile("file1.txt");
+        final FileObject file = resolveFile1Txt();
         assertTrue(file.exists());
         final FileObject folder = getReadFolderDir1();
         assertTrue(folder.exists());
@@ -265,6 +272,24 @@ public void testConcurrentReadFolder() throws Exception
         }
     }
 
+    public void testContent() throws Exception
+    {
+        final FileObject file = resolveFile1Txt();
+        assertTrue(file.exists());
+        final FileContent content = file.getContent();
+        assertNotNull(content);
+    }
+
+    public void testContentInfo() throws Exception
+    {
+        final FileObject file = resolveFile1Txt();
+        assertTrue(file.exists());
+        final FileContent content = file.getContent();
+        assertNotNull(content);
+        final FileContentInfo contentInfo = content.getContentInfo();
+        assertNotNull(contentInfo);
+    }
+    
     /**
      * Tests that findFiles() works.
      */
